api/rest/restapi_test.go

```diff
@@ -570,6 +571,76 @@ func TestAPIPinEndpoint(t *testing.T) {
 	testBothEndpoints(t, tf)
 }
 
+func TestAPIPinEndpointWithPath(t *testing.T) {
+	rest := testAPI(t)
+	defer rest.Shutdown()
+
+	testCases := []struct {
+		name    string
+		path    string
+		wantErr bool
+		code    int
+	}{
+		// Uncomment after removing logic Pin and Unpin with cid
```

Comment:
Not sure why this is commented out? And when it should be uncommented?

Reply:
With this PR, we will have two kind of APIs. Pin/Unpin with cid(what we already have) and Pin/Unpin with path(proposed in this PR). Since cid is a valid path, we don't really need separate handler for Pin/Unpin with cid. I am assuming that in future, we will get rid of that. That is the case when path is a cid.